esp/esp/program/modules/handlers/studentregphasezero.py

```diff
@@ -95,16 +93,15 @@ def studentregphasezero(self, request, tl, one, two, module, extra, prog):
         if 'op' in request.POST:
             op = request.POST['op']
         if op == 'join':
-            if len(request.POST['student_selected'].strip()) == 0:
+            if len(request.POST.get('student_selected', '').strip()) == 0:
                 join_error = 'Error - You must select a student\'s username.'
 
             else:
                 join_user = request.POST['student_selected']
-                group = PhaseZeroRecord.objects.filter(user=join_user, program=prog)[0]
+                group = PhaseZeroRecord.objects.get(user=join_user, program=prog)
```

Comment:
Wait, but in the `studentlookup` method, we only return students that are already in groups, so they shouldn't be able to provide a user that isn't in the lottery, right?